example/linalg/example_inverse3.f90

```diff
@@ -0,0 +1,23 @@
+! Matrix inversion example: in-place inversion
+program example_inverse3
+  use stdlib_linalg_constants, only: dp
+  use stdlib_linalg, only: invert,eye
+  implicit none
+
+  real(dp) :: A(2,2), Am1(2,2)
+
+  ! Input matrix (NB Fortran is column major! input columns then transpose)
+  A = transpose(reshape( [4, 3, &
+                          3, 2], [2,2] ))
+  Am1 = A
+
+  ! Invert matrix
+  call invert(Am1)
```

Comment:
Why not use this example to show how to use the subroutine interface with the optional output matrix:
```suggestion

  ! Invert matrix without modifying the original matrix
  Am1 = A
  call invert(Am1)
  ! Which would be equivalent to: call invert(A,Am1)
```

Reply:
[8b138f7](https://github.com/fortran-lang/stdlib/pull/828/commits/8b138f714761455dd7c787b5867b579dbb60e58f)